src/Support/RedisAdapter.php

```diff
@@ -159,18 +100,42 @@ public function xdel(string $stream, Collection|array $keys): int
      */
     public function pipeline(callable $closure): array
     {
-        if ($this->client() instanceof Pipeline) {
-            throw new RuntimeException('Pipelines are not able to be nested.');
-        }
-
         // Create a pipeline and wrap the Redis client in an instance of this class to ensure our wrapper methods are used within the pipeline...
         return $this->connection->pipeline(fn (Pipeline|PhpRedis $client) => $closure(new self($this->connection, $this->config, $client))); // @phpstan-ignore method.notFound
     }
 
     /**
-     * Get the Redis client instance.
+     * Run the given command.
+     */
+    protected function handle($args): mixed
+    {
+        try {
+            return tap($this->run($args), function ($result) {
+                if ($result === false && $this->client() instanceof PhpRedis) {
+                    throw new RedisClientException($this->client()->getLastError());
+                }
+            });
+        } catch (PredisServerException $e) {
+            throw new RedisClientException($e->getMessage(), previous: $e);
+        }
+    }
```

Comment:
This method ensures we have a consistent failure exception.

PhpRedis returns `false` on server error.
Predis throws a dedicated exception.